DeNaN: instrument parameters as they arrive in each function

kripken opened the pull request:
Instead of instrumenting every `local.get`, instrument parameters
on arrival at a function once on entry. After that, every local will
always contain a de-naned value (since we would denan on a
`local.set`). This is more efficient and also less confusing I think.

kripken commented on src/passes/DeNaN.cpp:
```diff
@@ -61,6 +67,33 @@ struct DeNaN : public WalkerPass<
+    }
+    if (!fixes.empty()) {
+      fixes.push_back(func->body);
+      func->body = builder.makeBlock(fixes);
```
We could run `--merge-blocks` perhaps to do that.

kripken commented:
I don't think we can exclude `local.tee` in general because it sets a new value, but maybe we can look into optimizing that pattern specifically.

aheejin commented:
Why not? Maybe I'm misunderstanding something... The expressions that had been already stored in the locals were deNaNed when they were stored, so we don't need to deNaN them when we load it again from it..?

kripken commented:
Oh sorry, I misunderstood. Yeah, since `tee` just lets a value fall through, without changing it, we can ignore it, very good point.

Added that now, and also added merge-blocks.
